scripts/algorithms/knn.dml

```diff
@@ -0,0 +1,745 @@
+#-------------------------------------------------------------
+# IBM Confidential
+# OCO Source Materials
+# ( C ) Copyright IBM Corp. 2010, 2015
+# The source code for this program is not published or
+# otherwise divested of its trade secrets, irrespective of
+# what has been deposited with the U.S. Copyright Office.
+#-------------------------------------------------------------
+#  
+# THIS SCRIPT IMPLEMENTS KNN( K Nearest Neighbor ) ALGORITHM
+#
+# INPUT   PARAMETERS:
+# ---------------------------------------------------------------------------------------------
+# NAME    TYPE     DEFAULT     OPTIONAL     MEANING
+# ---------------------------------------------------------------------------------------------
+# X                     String   ---   N    Location to read the input matrix as features
+# T                     String   ---   N    Location to read the input matrix for nearest neighbor
+#                                           search
+# X_T                   String   ---   Y    Location to read the feature type of matrix X an T
+#                                           whether columns in X are continuous ( =1 ) or
+#                                           categorical ( =2 )
+# Y                     String   ---    Y   Location to read the input matrix as target
+# Y_T                   String   ---    Y   Location to read the target type of matrix Y whether
+#                                           columns in Y are continuous ( =1 ) or
+#                                           categorical ( =2 )
+# K                     String   ---    Y   Location to write the result of selected k value
+# PR                    String   ---    Y   Location to write the predicted k-d tree search result,
```

Comment:
All parameters and comments related to k-d tree or hashing method should be removed.